cmd/kops/get_cluster.go

```diff
@@ -206,26 +214,52 @@ func clusterOutputTable(clusters []*api.Cluster, out io.Writer) error {
 	return t.Render(clusters, out, "NAME", "CLOUD", "ZONES")
 }
 
-func clusterOutputJson(clusters []*api.Cluster, out io.Writer) error {
-	for _, cluster := range clusters {
-		if err := marshalToWriter(cluster, marshalJSON, out); err != nil {
+// fullOutputJson outputs the marshalled JSON of a list of clusters and instance groups.  It will handle
+// nils for clusters and instanceGroups slices.
+func fullOutputJSON(out io.Writer, args ...runtime.Object) error {
+	argsLen := len(args)
+
+	if argsLen > 1 {
```

Comment:
I think we should always write the array, even if it is empty.  I don't think an empty string is valid JSON?

Reply:
JSON object is not an array. For instance when we create an ig, we just get a single object

Reply:
We can always write an array, but at least with a single JSON object create -f works, with a JSON array it bombs

Reply:
We discussed this; you are right - better to only write the array if there are multiple members.  This way a single object is an object